data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1138 @@
+
+local centrifuges = {}
+table.insert(centrifuges, aluminium_tubes)
+table.insert(centrifuges, machine_tools)
+table.insert(centrifuges, titanium)
+
+local custom_cargo = {
+	{ bkey = "CHEMICAL", goods = chemical, weight },
+	{ bkey = "MINING", goods = mining, weight },
+	{ bkey = "HARDWARE", goods = hardware, weight },
+	{ bkey = "INFRASTRUCTURE", goods = infrastructure, weight },
+	{ bkey = "CONSUMER_GOODS", goods = consumer_goods, weight },
+	{ bkey = "EXPENSIVE", goods = expensive, weight },
+	{ bkey = "FLUFFY", goods = fluffy, weight },
+	{ bkey = "WEDDING" , goods = wedding, weight },
+	{ bkey = "ART", goods = art, weight },
+	{ bkey = "GEMS", goods = gems, weight },
+	{ bkey = "RADIOACTIVE", goods = radioactive, weight },
+	{ bkey = "CENTRIFUGES", goods = centrifuges, weight }
+}
+
+-- Each branch should have a probability weight proportional to its size
+local custom_cargo_weight_sum = 0
+for branch,branch_array in pairs(custom_cargo) do
+	custom_cargo[branch].weight = #branch_array.goods
+	custom_cargo_weight_sum = custom_cargo_weight_sum + #branch_array.goods
+end
+
+local ads = {}
+local missions = {}
+
+local isQualifiedFor = function(reputation, ad)
+	return
+		reputation >= 12 or
+		ad.localdelivery or
+		(ad.risk <  0.1 and ad.urgency <= 0.1) or
+		(ad.risk <  0.5 and ad.urgency <= 0.5 and reputation >= 4) or
+		false
+end
+
+local getNumberOf = function (str)
+	local num = 0
+
+	while l[str .. "_" .. num + 1] do
+		num = num + 1
+	end
+	return num
+end
+
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+
+	form:Clear()
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	local qualified = isQualifiedFor(Character.persistent.player.reputation, ad)
+
+	form:SetFace(ad.client)
+
+	if not qualified then
+		form:SetMessage(l["DENY_" .. Engine.rand:Integer(1, getNumberOf("DENY"))])
+		return
+	end
+
+	if option == 0 then
+		local introtext = string.interp(ad.introtext, {
+			name		= ad.client.name,
+			cash		= Format.Money(ad.reward),
+			cargoname	= l[ad.cargotype.l10n_key],
+			starport	= ad.location:GetSystemBody().name,
+			system		= ad.location:GetStarSystem().name,
+			sectorx		= ad.location.sectorX,
+			sectory		= ad.location.sectorY,
+			sectorz		= ad.location.sectorZ,
+			dom_starport	= ad.domicile:GetSystemBody().name,
+			dom_system	= ad.domicile:GetStarSystem().name,
+			dom_sectorx	= ad.domicile.sectorX,
+			dom_sectory	= ad.domicile.sectorY,
+			dom_sectorz	= ad.domicile.sectorZ,
+			dist		= string.format("%.2f", ad.dist),
+		})
+		form:SetMessage(introtext)
+
+	elseif option == 1 then
+		local n = getNumberOf("WHYSOMUCH_" .. ad.branch)
+		if n >= 1 then
+			form:SetMessage(string.interp(l["WHYSOMUCH_" .. ad.branch .. "_" .. Engine.rand:Integer(1, n)], { cargoname = l[ad.cargotype.l10n_key] }))
+		elseif ad.urgency >= 0.8 then
+			form:SetMessage(string.interp(l["WHYSOMUCH_URGENT_" .. Engine.rand:Integer( 1, getNumberOf("WHYSOMUCH_URGENT"))], { cargoname = l[ad.cargotype.l10n_key] }))
```

Comment:
And again. For the next ones, I'll just assume you'll have them all fixed ;-)

(The rationale for that is that I consider the l10n_key to name relationship part of the equipment system internals)